mlrun/runtimes/databricks_runtime_wrapper.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import datetime
+import json
+import uuid
+from base64 import b64decode
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.compute import ClusterSpec
+from databricks.sdk.service.jobs import Run, SparkPythonTask, SubmitTask
+
+import mlrun
+
+
+def run_mlrun_databricks_job(
+    context,
+    mlrun_internal_code,
+    token_key="DATABRICKS_TOKEN",
+    timeout_minutes=20,
+    **kwargs,
+):
+
+    logger = context.logger
+    workspace = WorkspaceClient(token=mlrun.get_secret_or_env(key=token_key))
+    run_id = uuid.uuid4()
+    script_path_on_dbfs = (
+        f"/home/{workspace.current_user.me().user_name}/mlrun_databricks_runtime/"
+        f"mlrun_task_{run_id}.py"
+    )
+
+    code = b64decode(mlrun_internal_code).decode("utf-8")
+    with workspace.dbfs.open(script_path_on_dbfs, write=True, overwrite=True) as f:
+        f.write(code.encode("utf-8"))
+
+    def print_status(run: Run):
+        statuses = [f"{t.task_key}: {t.state.life_cycle_state}" for t in run.tasks]
+        logger.info(f'workflow intermediate status: {", ".join(statuses)}')
+
+    try:
+        cluster_id = mlrun.get_secret_or_env("DATABRICKS_CLUSTER_ID")
+        submit_task_kwargs = {}
+        if cluster_id:
+            logger.info(f"run with exists cluster_id: {cluster_id}")
+            submit_task_kwargs["existing_cluster_id"] = cluster_id
+        else:
+            logger.info("run with new cluster_id")
+            submit_task_kwargs["new_cluster"] = ClusterSpec(
+                spark_version=workspace.clusters.select_spark_version(
+                    long_term_support=True
+                ),
+                node_type_id=workspace.clusters.select_node_type(local_disk=True),
+                num_workers=1,
```

Comment:
Why are we fixing `num_workers=1`? Maybe set this as an optional parameter?

Reply:
will be fixed in following PR